regression-tests/test-results/pure2-enum.cpp

```diff
@@ -229,7 +229,7 @@ inline CPP2_CONSTEXPR file_attributes file_attributes::none = 0;
     if (((*this) & current) == current) {_ret += _comma + "current";_comma = ", ";}
     if (((*this) & obsolete) == obsolete) {_ret += _comma + "obsolete";_comma = ", ";}
     if (((*this) & cached_and_current) == cached_and_current) {_ret += _comma + "cached_and_current";_comma = ", ";}
-    return _ret + ")"; 
+    return std::move(_ret) + ")"; 
```

Comment:
This doesn't look like a good place to have `std::move`.

Reply:
We definitely want to move here.
Consider if `_ret`'s size was `std::string::max_size() - 1`.
You avoid copying all that.

Reply:
Oh, but probably not in this specific context of an enumerator name.
Should I discard `_ret` after the return statement to avoid the move?

Reply:
Moving here is automatic.  You don't need to tell the compiler to move.  Adding `std::move()` is a pessimization.  https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2991r0.html

Reply:
Ah, it's a combination of `_ret` and a string,, so you probably just want to do `_ret += ")"; return _ret;`

Reply:
But since this is a transform of existing non-optimal code, I guess it does make sense.

Reply:
That isn't adopted, and wouldn't apply here anyways.
Clang and GCC have `-Wpessimize-move`.
You could try to craft an example that triggers on `return std::move(_ret) + ")"; `.

Reply:
A good implementation would make the implicit move and explicit `+=` equivalent.
And then leave it in the hands of a good optimizer to make them really equal.
Which might itself rely on the branch predictor to make them equally performant.

Reply:
I know it's not adopted, just has the rationale about why you shouldn't move directly into a return, but that wasn't actually what this was doing.  Too many strings around, I didn't fully understand what it was doing at first.

Reply:
At some point, someone pointed out that cppfront was doing it wrong,
and now it doesn't add unnecessary moves and relies on NRVO.
There's just one case: <https://github.com/hsutter/cppfront/commit/13f765ba7927478162318325eff55b2a01c3da81#commitcomment-133825753>.

Reply:
So really, it's an issue with this line:
https://github.com/hsutter/cppfront/blob/main/source/reflect.h2#L1022
which would be better as
```
to_string += "    _ret += ")"; return _ret;\n}\n";
```
so it avoids the need to insert the move.  Maybe that's an optimization that compilers can do, but it seems unlikely.